Library/Formula/poco.rb

```diff
@@ -22,9 +22,19 @@ class Poco < Formula
   depends_on "openssl"
 
   def install
+    #This modifies the install name of shared libraries so the path is the install directory instead of the build directory.
+    #https://github.com/pocoproject/poco/pull/670
+    files = ['build/config/Darwin-clang','build/config/Darwin-clang-libc++','build/config/Darwin-gcc']
+    inreplace files, '-dynamiclib', '-dynamiclib -Wl,-install_name,$(POCO_PREFIX)/lib/$(notdir \$@)'
+
     ENV.cxx11 if build.cxx11?
 
-    arch = Hardware.is_64_bit? ? 'Darwin64': 'Darwin32'
+    if MacOS.version < :mavericks
+      arch = Hardware.is_64_bit? ? 'Darwin64': 'Darwin32'
+    else
+      arch = Hardware.is_64_bit? ? 'Darwin64-clang-libc++': 'Darwin32-clang-libc++'
```

Comment:
Need to branch this off of `ENV.compiler`, too - users could still choose to use GCC.